### ISSUE TITLE
ConcurrencyError: No message available

### ISSUE THREAD

**Issue by McStork** [logstash-plugins/logstash-input-beats#14]
Hello,
I am a user of Packetbeat trying to improve Logstash throughput. I'm using this guide: https://www.elastic.co/blog/logstash-configuration-tuning

The first objective would be to save the packets in a JSON file. Here is the Logstash config file:

```
input {
  beats {
    port => 5044
  }
}

output {
  file {
    codec => "json_lines"
    path => "/opt/packetbeat_shipper/packets_test.json"
  }
}
```

The following error message is what I get with this config file.

`
{:timestamp=>"2015-11-04T09:24:48.686000+0100", :message=>"Beats input: unhandled exception", :exception=>#<ConcurrencyError: No message available>, :backtrace=>["org/jruby/ext/thread/ConditionVariable.java:98:in `wait'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-input-lumberjack-2.0.4/lib/logstash/sized_queue_timeout.rb:30:in `push'", "org/jruby/ext/thread/Mutex.java:149:in `synchronize'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-input-lumberjack-2.0.4/lib/logstash/sized_queue_timeout.rb:27:in `push'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-input-beats-0.9.3/lib/logstash/inputs/beats.rb:110:in `run'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-input-lumberjack-2.0.4/lib/logstash/circuit_breaker.rb:42:in `execute'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-input-beats-0.9.3/lib/logstash/inputs/beats.rb:109:in `run'", "org/jruby/RubyProc.java:271:in `call'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-input-beats-0.9.3/lib/logstash/inputs/beats.rb:167:in `invoke'", "org/jruby/RubyProc.java:271:in `call'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-input-beats-0.9.3/lib/lumberjack/beats/server.rb:378:in `data'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-input-beats-0.9.3/lib/lumberjack/beats/server.rb:357:in `read_socket'", "org/jruby/RubyProc.java:271:in `call'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-input-beats-0.9.3/lib/lumberjack/beats/server.rb:369:in `ack_if_needed'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-input-beats-0.9.3/lib/lumberjack/beats/server.rb:353:in `read_socket'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-input-beats-0.9.3/lib/lumberjack/beats/server.rb:246:in `json_data_payload'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-input-beats-0.9.3/lib/lumberjack/beats/server.rb:163:in `feed'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-input-beats-0.9.3/lib/lumberjack/beats/server.rb:296:in `compressed_payload'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-input-beats-0.9.3/lib/lumberjack/beats/server.rb:163:in `feed'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-input-beats-0.9.3/lib/lumberjack/beats/server.rb:338:in `read_socket'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-input-beats-0.9.3/lib/lumberjack/beats/server.rb:315:in `run'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-input-beats-0.9.3/lib/logstash/inputs/beats.rb:167:in `invoke'", "org/jruby/RubyProc.java:271:in `call'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/concurrent-ruby-0.9.1-java/lib/concurrent/executor/executor_service.rb:515:in `run'", "Concurrent$$JavaExecutorService$$Job_1833200224.gen:13:in `run'"], :level=>:error}
`

**Comment by urso** [comment]
which logstash version are you using?

For some reason the worker thread handling the connection has been interrupted. The Java [InterruptedException](http://docs.oracle.com/javase/7/docs/api/java/lang/InterruptedException.html) is thrown.

Did the exception occur when stopping logstash?

**Comment by ph** [comment]
@McStork Do you have a lot of clients connection to the logstash side?

**Comment by McStork** [comment]
Logstash 2.0
@ph No, only one client

**Comment by ph** [comment]
@McStork JDK / OS version?

**Comment by McStork** [comment]
openjdk version "1.8.0_66-internal"
Debian 8.2

**Comment by McStork** [comment]
It works now. I'm sorry, I hope you didn't spend too much time on this. The exception was in the .log output file and I missed what was inside the .err file... Errno::EACCES: Permission denied - /opt/packetbeat_shipper/packets_test.json.

**Comment by ph** [comment]
I think this issue is really similar to https://github.com/logstash-plugins/logstash-input-beats/issues/37
event if the message is a bit different.

**Comment by ph** [comment]
have release 2.1.0 of this plugins, that should fix the ConcurrencyError issue.

**Comment by ph** [comment]
#37 should fix this released in 2.1.1, if you still see this error reopen this issue.

**Comment by McStork** [comment]
@ph Good job, thanks!
